src/prefect/utilities/tasks.py

```diff
@@ -8,13 +10,170 @@
 
 import prefect
 
-__all__ = ["tags", "as_task", "pause_task", "task", "unmapped", "defaults_from_attrs"]
+__all__ = [
+    "tags",
+    "as_task",
+    "pause_task",
+    "task",
+    "unmapped",
+    "apply_map",
+    "defaults_from_attrs",
+]
 
 if TYPE_CHECKING:
     import prefect.tasks.core.constants
     import prefect.tasks.core.collections
     import prefect.tasks.core.function
-    from prefect.core.flow import Flow
+    from prefect import Flow
+
+
+def apply_map(func: Callable, *args: Any, flow: "Flow" = None, **kwargs: Any) -> Any:
+    """
+    Map a function that adds tasks to a flow elementwise across one or more
+    tasks.  Arguments that should _not_ be mapped over should be wrapped with
+    `prefect.unmapped`.
+
+    This can be useful when wanting to create complicated mapped pipelines
+    (e.g. ones using control flow components like `case`).
+
+    Args:
+        - func (Callable): a function that adds tasks to a flow
+        - *args: task arguments to map over
+        - flow (Flow, optional): The flow to use, defaults to the current flow
+            in context if no flow is specified. If specified, `func` must accept
+            a `flow` keyword argument.
+        - **kwargs: keyword task arguments to map over
+
+    Returns:
+        - Any: the output of `func`, if any
+
+    Example:
+
+    ```python
+    from prefect import task, case, apply_map
+    from prefect.tasks.control_flow import merge
+
+    @task
+    def inc(x):
+        return x + 1
+
+    @task
+    def is_even(x):
+        return x % 2 == 0
+
+    def inc_if_even(x):
+        with case(is_even(x), True):
+            x2 = inc(x)
+        return merge(x, x2)
+
+    with Flow("example") as flow:
+        apply_map(inc_if_even, range(10))
+    """
+    from prefect.tasks.core.constants import Constant
+
+    no_flow_provided = flow is None
+    if no_flow_provided:
+        flow = prefect.context.get("flow", None)
+        if flow is None:
+            raise ValueError("Couldn't infer a flow in the current context")
+    assert isinstance(flow, prefect.Flow)  # appease mypy
+
+    # Check if args/kwargs are valid first
+    for x in itertools.chain(args, kwargs.values()):
+        if not isinstance(x, (prefect.Task, unmapped, Sequence)):
+            raise TypeError(
+                f"Cannot map over non-sequence object of type `{type(x).__name__}`"
+            )
+
+    flow2 = prefect.Flow("temporary flow")
+    # A mapping of all the input args -> (is_mapped, is_constant)
+    arg_info = {}
+    # A mapping of the ids of all constants -> the Constant task.
+    # Used to convert constants to constant tasks if needed
+    id_to_const = {}
+
+    def preprocess(a: Any) -> "prefect.Task":
```

Comment:
Could you add a comment to this block (L95-L107) that this is where the inputs to `apply_map` are added to the flow?

Reply:
Done.